oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsHWDiskStore.java

```diff
@@ -277,29 +278,49 @@ private static PartitionMaps queryPartitionMaps() {
             mAnt = DEVICE_ID.matcher(WmiUtil.getRefString(diskPartitionMap, DiskToPartitionProperty.ANTECEDENT, i));
             mDep = DEVICE_ID.matcher(WmiUtil.getRefString(diskPartitionMap, DiskToPartitionProperty.DEPENDENT, i));
             if (mAnt.matches() && mDep.matches()) {
-                maps.partitionToLogicalDriveMap.put(mAnt.group(1), mDep.group(1) + "\\");
+               if(maps.partitionToLogicalDriveMap.containsKey(mAnt.group(1))) {
+                   maps.partitionToLogicalDriveMap.get(mAnt.group(1)).add(mDep.group(1)+ "\\");
+               } else {
+                  List<String>  list  = new LinkedList<String>();
```

Comment:
hi, I used ArrayList instead of LinkedList.  I try to use Win32_LogicalDiskToPartition to get logical partition SIZE, but I don't know how to get it to work. The good news is I found  another solution,  I used Kernel32.INSTANCE.GetDiskFreeSpaceEx(...) to get volume SIZE, and I used C on Visual Studio to test method GetDiskFreeSpaceEx, they both have the same result. This is what I can do now, I learned a lot from oshi. Thanks! If I find anything else I will let you know, and If you find something wrong  please let me know.

Reply:
> I try to use Win32_LogicalDiskToPartition to get logical partition SIZE, but I don't know how to get it to work.

You should be able to modify the Enum in the class `oshi.driver.windows.wmi.Win32LogicalDiskToPartition` which currently only includes `ANTECEDENT` and `DEPENDENT`.  Include the additional two elements `ENDINGADDRESS` and `STARTINGADDRESS`.  These will then be available for you after the query, and you can get the values with `WmiUtil.getUint64()`.  Ending address plus 1 minus starting address should give you size.

If you can provide me the `wmic path Win32_LogicalDiskToPartition` I can make the changes here as well, I just need to understand what your system is showing you for that.

Reply:
I just added a commit to your branch that calculates the size from WMI.  It's slightly off from your output, but I think it's more accurate for partition size.  

Here's the current debug output on my Win7 VM's single partition:
`CHECK: volumeSize=68717375488, wmi address calulated size=68717379584`

But the overall disk partition size matches the wmi calculation:
`Disk partition size = 68717379584`

Can you check this on your system and see if it's good?  If so, remove the FIXME stuff. :)

Reply:
(The difference in your `volumeSize` and the WMI calculation is exactly 4096 bytes, probably 8 512K blocks of partition header info that is not available to the logical volume but is part of the allocated size.)

Reply:
Ok, I  post the output of all my logs here about hwPartitionQueryMap  from
   WmiResult<DiskPartitionProperty> hwPartitionQueryMap = Win32DiskPartition.queryPartition();
getCIMType  ANTECEDENT:102
getCIMType  DEPENDENT:102
getCIMType  STARTINGADDRESS:21
getCIMType  ENDINGADDRESS:21
    getVtType  ANTECEDENT:8
    getVtType  DEPENDENT:8
    getVtType  STARTINGADDRESS:8
    getVtType  ENDINGADDRESS:8
       getValue i=0, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #0, Partition #2"
       getValue i=0, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="C:"
       getValue i=0, STARTINGADDRESS:960495616
       getValue i=0, ENDINGADDRESS:96637812735

       getValue i=1, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #0, Partition #3"
       getValue i=1, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="D:"
       getValue i=1, STARTINGADDRESS:96637812736
       getValue i=1, ENDINGADDRESS:257699086335

       getValue i=2, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #0, Partition #4"
       getValue i=2, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="E:"
       getValue i=2, STARTINGADDRESS:257699086336
       getValue i=2, ENDINGADDRESS:472447451135

       getValue i=3, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #0, Partition #5"
       getValue i=3, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="F:"
       getValue i=3, STARTINGADDRESS:472447451136
       getValue i=3, ENDINGADDRESS:601296470015

       getValue i=4, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #0, Partition #6"
       getValue i=4, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="G:"
       getValue i=4, STARTINGADDRESS:601296470016
       getValue i=4, ENDINGADDRESS:654984609791

       getValue i=5, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #1, Partition #0"
       getValue i=5, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="I:"
       getValue i=5, STARTINGADDRESS:1048576
       getValue i=5, ENDINGADDRESS:144955146239

       getValue i=6, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #1, Partition #1"
       getValue i=6, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="J:"
       getValue i=6, STARTINGADDRESS:144956194816
       getValue i=6, ENDINGADDRESS:230855540735

       getValue i=7, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #1, Partition #2"
       getValue i=7, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="K:"
       getValue i=7, STARTINGADDRESS:230856589312
       getValue i=7, ENDINGADDRESS:338230771711

       getValue i=8, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #1, Partition #2"
       getValue i=8, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="L:"
       getValue i=8, STARTINGADDRESS:338231820288
       getValue i=8, ENDINGADDRESS:552980185087

       getValue i=9, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #1, Partition #2"
       getValue i=9, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="M:"
       getValue i=9, STARTINGADDRESS:552981233664
       getValue i=9, ENDINGADDRESS:875103780863

       getValue i=10, ANTECEDENT: \\DESKTOP-7KF7BMC\root\cimv2:Win32_DiskPartition.DeviceID="Disk #1, Partition #2"
       getValue i=10, DEPENDENT:\\DESKTOP-7KF7BMC\root\cimv2:Win32_LogicalDisk.DeviceID="N:"
       getValue i=10, STARTINGADDRESS:875104829440
       getValue i=10, ENDINGADDRESS:1000169537535

deviceID:Disk #1, Partition #0, logicalDrive：I:\,  GetDiskFreeSpaceEx size:144954093568,  wmi address calulated size: 144954097664, diff:4096
deviceID:Disk #1, Partition #1, logicalDrive：J:\,  GetDiskFreeSpaceEx size:85899341824,  wmi address calulated size: 85899345920, diff:4096
deviceID:Disk #1, Partition #2, logicalDrive：K:\,  GetDiskFreeSpaceEx size:107374178304,  wmi address calulated size: 107374182400, diff:4096
deviceID:Disk #1, Partition #2, logicalDrive：L:\,  GetDiskFreeSpaceEx size:214748360704,  wmi address calulated size: 214748364800, diff:4096
deviceID:Disk #1, Partition #2, logicalDrive：M:\,  GetDiskFreeSpaceEx size:322122543104,  wmi address calulated size: 322122547200, diff:4096
deviceID:Disk #1, Partition #2, logicalDrive：N:\,  GetDiskFreeSpaceEx size:125064704000,  wmi address calulated size: 125064708096, diff:4096
deviceID:Disk #0, Partition #2, logicalDrive：C:\,  GetDiskFreeSpaceEx size:95677313024,  wmi address calulated size: 95677317120, diff:4096
deviceID:Disk #0, Partition #3, logicalDrive：D:\,  GetDiskFreeSpaceEx size:161061269504,  wmi address calulated size: 161061273600, diff:4096
deviceID:Disk #0, Partition #4, logicalDrive：E:\,  GetDiskFreeSpaceEx size:214748360704,  wmi address calulated size: 214748364800, diff:4096
deviceID:Disk #0, Partition #5, logicalDrive：F:\,  GetDiskFreeSpaceEx size:128849014784,  wmi address calulated size: 128849018880, diff:4096
deviceID:Disk #0, Partition #6, logicalDrive：G:\,  GetDiskFreeSpaceEx size:53688135680,  wmi address calulated size: 53688139776, diff:4096

    I saw you code and know what you meant and I ran your code on windows 10. I use GetDiskFreeSpaceEx  get partition SIZE bigger than (wmi address calulated) SIZE for 4096 , I'm not sure which  one is more accurate. I used C on Visual Studio 2019 to  execute method GetDiskFreeSpaceEx , it gets same results like Kernel32.INSTANCE.GetDiskFreeSpaceEx. I didn't think so much before , now I am confused which one is better. 
    Above log is all my HARD DRIVE infomation, maybe it can help you, Thanks man, I love this project.

Reply:
OK, great! It looks like my size calculation works (with the constant 4096 offset.)

I'll remove the debug code and merge.

Reply:
I think the higher number is the "hardware partition size" and the number smaller by 4096 bytes is the available size to software (free space).  The 4096 bytes is probably header/mapping/etc. info that the software can read.  But the higher number is consistent with the other "partition size" values so I trust it.

Reply:
Yep, looks like the 4096 is comprized of the LBA header, which is about 62 bytes of actual data and a bunch of zeroes to stay aligned to 4096-byte sectors. :)

Reply:
Thanks man, you did very good job. This project help me a lot, I'm gonna pull the code now